asciimatics/screen.py

```diff
@@ -2508,6 +2508,7 @@ def get_event(self):
             """
             # Spin through notifications until we find something we want.
             key = 0
+            is_utf_char = False
```

Comment:
I'm nervous that this will hit potential bugs if the input stream gets split (e.g. because of network delays).  The rest of this function tries to maintain state between calls to handle this.  

Could you make this an internal property of the screen to handle this (like `_bytes_to_read`)?

Reply:
I used `_bytes_to_read` value as an indicator whether we're dealing with the utf-8 characters and after every character this value is being reset to the default one.